metapool/controls.py

```diff
@@ -158,6 +159,38 @@ def get_controls_details_from_context(sample_context):
     return result
 
 
+def denormalize_controls_details(controls_details):
+    denormalized_controls_details = []
+    if controls_details is None:
+        return None
+
+    def _denormalize_details_record(a_record, qiita_id):
+        # remove the primary and secondary studies entries from the record,
+        # replace with a single QIITA_ID_KEY entry; leave everything else alone
+        denormalized_record = a_record.copy()
+        denormalized_record.pop(SECONDARY_STUDIES_KEY)
+        denormalized_record.pop(PRIMARY_STUDY_KEY)
+        denormalized_record[QIITA_ID_KEY] = qiita_id
+        return denormalized_record
+
+    for curr_record in controls_details.values():
+        for curr_secondary_study in curr_record[SECONDARY_STUDIES_KEY]:
+            curr_denorm_secondary_record = _denormalize_details_record(
+                curr_record, curr_secondary_study)
+            denormalized_controls_details.append(curr_denorm_secondary_record)
+        # next secondary study, if any
+
+        curr_denorm_primary_record = _denormalize_details_record(
+            curr_record, curr_record[PRIMARY_STUDY_KEY])
+        denormalized_controls_details.append(curr_denorm_primary_record)
```

Comment:
Just to confirm, does this mean that we only support up to 2 studies?

Reply:
I don't think that's what the code shows, as there is an iteration over `curr_record[SECONDARY_STUDIES_KEY]`

Reply:
We are not limited to 2; there can be an arbitrary number of secondary studies :)